src/NovelAudioChannel.cpp

```diff
@@ -0,0 +1,82 @@
+//
+// Created by krjoh on 29/08/2019.
+//
+
+#include "NovelAudioChannel.h"
+#include <string>
+#include <map>
+
+namespace NovelRT {
+NovelAudioChannel::NovelAudioChannel(std::string input, bool isBgm, int channel){
+  if (isBgm)
+  {
+    _loadFile(input, 0);
+    _isMusic = true;
+    _channel = 0;
+  }
+  else
+  {
+    _loadFile(input, channel);
+    _isMusic = false;
+    _channel = channel;
+  }
+}
+
+bool NovelAudioChannel::_loadFile(std::string in, int channelAssigned){
+  if (!channelAssigned == 0)
+  {
+    _effect = Mix_LoadWAV(in.c_str());
+    if (_effect == NULL)
+    {
+      std::cout << Mix_GetError() << std::endl;
+      return false;
+    }
+  }
+  else
+  {
+    _bgm = Mix_LoadMUS(in.c_str());
+    if (_bgm == NULL)
+    {
+      std::cout << Mix_GetError() << std::endl;
+      return false;
+    }
+  }
+  return true;
+}
+
+Mix_Chunk* NovelAudioChannel::getChunk(){
+  return _effect;
+}
+
+void NovelAudioChannel::play()
+{
+  if (_isMusic)
+  {
+  if (Mix_PlayingMusic() == 0)
+  {
+    Mix_PlayMusic(_bgm, 0);
+    std::cout << "Playing music on Channel 0!" << std::endl;
+  }
+  else
+  {
+    if (Mix_PausedMusic() == 1)
+    {
+      Mix_ResumeMusic();
+    }
+    else
+    {
+      Mix_PauseMusic();
+    }
+  }
+  }
+  else
+  {
+    if (Mix_Playing(_channel) == 0)
```

Comment:
With all these numbers im wondering if we can't begin treating this more like `bool` rather than `int`.